src/v/security/audit/audit_log_manager.h

```diff
@@ -371,6 +371,7 @@ class audit_log_manager
     /// Other references
     cluster::controller* _controller;
     kafka::client::configuration& _config;
+    bool _recovery_mode_enabled{false};
```

Comment:
this isn't necessary you can just check the recovery_mode_enabled config value. 

A bit nitpicky but a bool member variable that isn't const implies it can be modified, and toggled which this isn't so I'd rather directly refer to the config value